modules/dists/DSIUtil.chpl

```diff
@@ -647,6 +647,22 @@ proc bulkCommTranslateDomain(srcSlice : domain, srcDom : domain, targetDom : dom
   return {(...rngs)};
 }
 
+// this is an overload for views that are ranges. This will be used by code
+// paths that comes from elided array views. We could consider using this
+// lighterweight implementation as a more general special case for 1D bulk
+// transfers.
+proc bulkCommTranslateDomain(srcSlice : domain, srcView : range(?),
+                             targetView : range(?)) {
+  if srcSlice.rank != 1 then
+    compilerError("bulkCommTranslateDomain: source slice and source domain must have identical rank");
+
+  param strides = chpl_strideUnion(targetView, srcSlice);
+
+  const dense = densify(srcSlice.dim(0), srcView);
+  const rngs  = unDensify(dense, targetView);
+
+  return {rngs};
```

Comment:
I'm probably missing something here, but assuming this is supposed to return a domain, shouldn't this line be `return {(...rngs)};`?

Reply:
`rngs` is actually just a range. The variable name is the same as the `rngs` in the original version, which as you'd expect is a tuple of ranges. I'll fix the variable name.